drivers/ethernet/eth_xlnx_gem.c

```diff
@@ -0,0 +1,1724 @@
+static void eth_xlnx_gem_handle_tx_done(const struct device *dev);
+
+static const struct ethernet_api eth_xlnx_gem_apis = {
+	.iface_api.init		= eth_xlnx_gem_iface_init,
+	.get_capabilities	= eth_xlnx_gem_get_capabilities,
+	.send			= eth_xlnx_gem_send,
+	.start			= eth_xlnx_gem_start_device,
+	.stop			= eth_xlnx_gem_stop_device,
+#if defined(CONFIG_NET_STATISTICS_ETHERNET)
+	.get_stats		= eth_xlnx_gem_stats,
+#endif
+};
+
+#if DT_NODE_HAS_STATUS(DT_DRV_INST(0), okay)
+
+/* IRQ configuration function for GEM0 */
+ETH_XLNX_GEM_CONFIG_IRQ_FUNC(0);
+/* GEM0 device configuration data */
+ETH_XLNX_GEM_DEV_CONFIG(0);
+/* GEM0 run-time device data */
+ETH_XLNX_GEM_DEV_DATA(0);
+/* GEM0 DMA memory area */
+ETH_XLNX_GEM_DMA_AREA_DECL(0);
+ETH_XLNX_GEM_DMA_AREA_INST(0);
+/* GEM0 driver instance declaration */
+ETH_XLNX_GEM_NET_DEV_INIT(0);
+
+#endif /* DT_NODE_HAS_STATUS(DT_DRV_INST(0), okay) */
+
+#if DT_NODE_HAS_STATUS(DT_DRV_INST(1), okay)
+
+/* IRQ configuration function for GEM1 */
+ETH_XLNX_GEM_CONFIG_IRQ_FUNC(1);
+/* GEM1 device configuration data */
+ETH_XLNX_GEM_DEV_CONFIG(1);
+/* GEM1 run-time device data */
+ETH_XLNX_GEM_DEV_DATA(1);
+/* GEM1 DMA memory area */
+ETH_XLNX_GEM_DMA_AREA_DECL(1);
+ETH_XLNX_GEM_DMA_AREA_INST(1);
+/* GEM1 driver instance declaration */
+ETH_XLNX_GEM_NET_DEV_INIT(1);
+
+#endif /* DT_NODE_HAS_STATUS(DT_DRV_INST(1), okay) */
+
+#if DT_NODE_HAS_STATUS(DT_DRV_INST(2), okay)
+
+/* IRQ configuration function for GEM2 */
+ETH_XLNX_GEM_CONFIG_IRQ_FUNC(2);
+/* GEM2 device configuration data */
+ETH_XLNX_GEM_DEV_CONFIG(2);
+/* GEM2 run-time device data */
+ETH_XLNX_GEM_DEV_DATA(2);
+/* GEM2 DMA memory area */
+ETH_XLNX_GEM_DMA_AREA_DECL(2);
+ETH_XLNX_GEM_DMA_AREA_INST(2);
+/* GEM2 driver instance declaration */
+ETH_XLNX_GEM_NET_DEV_INIT(2);
+
+#endif /* DT_NODE_HAS_STATUS(DT_DRV_INST(2), okay) */
+
+#if DT_NODE_HAS_STATUS(DT_DRV_INST(3), okay)
+
+/* IRQ configuration function for GEM3 */
+ETH_XLNX_GEM_CONFIG_IRQ_FUNC(3);
+/* GEM3 device configuration data */
+ETH_XLNX_GEM_DEV_CONFIG(3);
+/* GEM3 run-time device data */
+ETH_XLNX_GEM_DEV_DATA(3);
+/* GEM3 DMA memory area */
+ETH_XLNX_GEM_DMA_AREA_DECL(3);
+ETH_XLNX_GEM_DMA_AREA_INST(3);
+/* GEM3 driver instance declaration */
+ETH_XLNX_GEM_NET_DEV_INIT(3);
+
+#endif /* DT_NODE_HAS_STATUS(DT_DRV_INST(3), okay) */
+
+/**
+ * @brief GEM device initialization function
+ * Initializes the GEM itself, the DMA memory area used by the GEM and,
+ * if enabled, an associated PHY attached to the GEM's MDIO interface.
+ *
+ * @param dev Pointer to the device data
+ * @retval 0 if the device initialization completed successfully
+ */
+static int eth_xlnx_gem_dev_init(const struct device *dev)
+{
+	const struct eth_xlnx_gem_dev_cfg *dev_conf = DEV_CFG(dev);
+	uint32_t reg_val = 0;
+
+	/* Precondition checks using assertions */
+
+	/* Valid PHY address and polling interval, if PHY is to be managed */
+	if (dev_conf->init_phy == 1) {
```

Comment:
I just wonder if it would be more clearer if `init_phy` is made boolean as it seems to be either 0 or 1. Then the reader would not need to guess if the integer value has some special meaning, and we could write here
`if (dev_conf->init_phy) {`

Reply:
All device configuration parameters of type uint8_t with boolean semantics have been changed to type bool. All if-statements evaluating them have been changed to the syntax you suggested.